ratis-server/src/main/java/org/apache/ratis/server/impl/StateMachineUpdater.java

```diff
@@ -190,11 +189,10 @@ public void run() {
           reload();
         }
 
-        final MemoizedSupplier<List<CompletableFuture<Message>>> futures = applyLog();
-        checkAndTakeSnapshot(futures);
+        applyLogFutures = applyLog(applyLogFutures);
```

Comment:
Let's check if `applyLogFutures.isCompletedExceptionally()` before applying logs.

Reply:
But in this case we have already submitted next set of tasks right? What would be the point of checking the exception?

Reply:
The point is to fail fast.  Otherwise, it will keep applying log even if the state machine has failed.

Reply:
The statemachine can take a call right if it has failed. From what I understand statemachine should receive all the transactions, and it is on the statemachine to have the guardrail if it should apply a transaction or not.

Reply:
If we want to fail fast we should fail the group remove as well.